vnext/Microsoft.ReactNative/JsiApi.cpp

```diff
@@ -623,124 +696,131 @@ JsiHostFunction JsiRuntime::GetHostFunction(JsiObjectRef func) try {
+  auto propertyIdPtr = RuntimeAccessor::AsPointerValue(propertyId);
+  return m_runtimeAccessor->hasProperty(
+      RuntimeAccessor::AsObject(&objPtr), RuntimeAccessor::AsPropNameID(&propertyIdPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 void JsiRuntime::SetProperty(JsiObjectRef obj, JsiPropertyIdRef propertyId, JsiValueRef const &value) try {
-  auto objPtr = AsPointerValue(obj);
-  auto propertyIdPtr = AsPointerValue(propertyId);
-  m_runtime->setPropertyValue(
-      const_cast<facebook::jsi::Object &>(AsObject(&objPtr)), AsPropNameID(&propertyIdPtr), *AsValue(value));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  auto propertyIdPtr = RuntimeAccessor::AsPointerValue(propertyId);
+  m_runtimeAccessor->setPropertyValue(
+      const_cast<facebook::jsi::Object &>(RuntimeAccessor::AsObject(&objPtr)),
+      RuntimeAccessor::AsPropNameID(&propertyIdPtr),
+      *RuntimeAccessor::AsValue(value));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 JsiObjectRef JsiRuntime::GetPropertyIdArray(JsiObjectRef obj) try {
-  auto objPtr = AsPointerValue(obj);
-  return MakeJsiArrayData(m_runtime->getPropertyNames(AsObject(&objPtr)));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  return PointerAccessor::MakeJsiArrayData(m_runtimeAccessor->getPropertyNames(RuntimeAccessor::AsObject(&objPtr)));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 bool JsiRuntime::IsArray(JsiObjectRef obj) try {
-  auto objPtr = AsPointerValue(obj);
-  return m_runtime->isArray(AsObject(&objPtr));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  return m_runtimeAccessor->isArray(RuntimeAccessor::AsObject(&objPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 bool JsiRuntime::IsArrayBuffer(JsiObjectRef obj) try {
-  auto objPtr = AsPointerValue(obj);
-  return m_runtime->isArrayBuffer(AsObject(&objPtr));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  return m_runtimeAccessor->isArrayBuffer(RuntimeAccessor::AsObject(&objPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 bool JsiRuntime::IsFunction(JsiObjectRef obj) try {
-  auto objPtr = AsPointerValue(obj);
-  return m_runtime->isFunction(AsObject(&objPtr));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  return m_runtimeAccessor->isFunction(RuntimeAccessor::AsObject(&objPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 bool JsiRuntime::IsHostObject(JsiObjectRef obj) try {
-  auto objPtr = AsPointerValue(obj);
-  return m_runtime->isHostObject(AsObject(&objPtr));
+  auto objPtr = RuntimeAccessor::AsPointerValue(obj);
+  return m_runtimeAccessor->isHostObject(RuntimeAccessor::AsObject(&objPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
 
 bool JsiRuntime::IsHostFunction(JsiObjectRef func) try {
-  auto funcPtr = AsPointerValue(func);
-  return m_runtime->isHostFunction(AsFunction(&funcPtr));
+  auto funcPtr = RuntimeAccessor::AsPointerValue(func);
+  return m_runtimeAccessor->isHostFunction(RuntimeAccessor::AsFunction(&funcPtr));
 } catch (JSI_SET_ERROR) {
   throw;
 }
```

Comment:
can all these empty catches and their corresponding try-s be removed? (since they aren't doing anything)

Reply:
No, they must not be removed. They are a key to proper exception handling. Please have a look at the JSI_SET_ERROR. It has special handling for facebook::jsi::JSError and facebook::jsi::JSINativeException exceptions. It was hidden in the macro to simplify the code look, but I still want to show that we handle exceptions.